packages/lesswrong/server/migrations/2020-05-06-forceSecureImageLinks.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Turn insecure image links into secure ones if the secure image exists. We
+ * only edit the html (not the original content), so it's possible someone will
+ * edit their post and the update will be reverted.
+ */
+
+import { registerMigration, forEachDocumentBatchInCollection } from './migrationUtils';
+import { urlIsBroken } from '../scripts/utils'
+import { Posts } from '../../lib/collections/posts/collection';
+import cheerio from 'cheerio'
+
+function findInsecureImages ($: any): Array<string> {
+  let insecureImageSources: Array<string> = []
+  $('img').each(function (this: any) {
+    const el = $(this)
+    const src = el.attr('src') as string
+    const testResult = /^http:\/\//.test(src)
+    if (testResult) {
+      insecureImageSources.push(src)
+    }
+  })
+  return insecureImageSources
+}
+
+async function testSecureImages
```

Comment:
I think some docstrings would be good, especially because the function names are somewhat confusingly similar.